### PR TITLE
Run openshift permission tests in parallel

### PR DESCRIPTION
**Pull request by rmohr**
**What this PR does / why we need it**:

Run 14 permission tests in parallel. Every test gets its own unique user.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/access_test.go [hunk 1]
```diff
@@ -345,16 +345,19 @@ var _ = Describe("[rfe_id:500][crit:high][arm64][vendor:cnv-qe@redhat.com][level
...
 	})
 
-	Describe("[Serial][rfe_id:2919][crit:high][vendor:cnv-qe@redhat.com][level:component] With regular OpenShift user", func() {
+	Describe("[rfe_id:2919][crit:high][vendor:cnv-qe@redhat.com][level:component] With regular OpenShift user", func() {
+
+		var testUser string
+
 		BeforeEach(func() {
+			// Generate unique usernames based on the test namespace which is unique per ginkgo node
+			testUser = "testuser-" + util.NamespaceTestDefault
```
Comment (xpivarc) [inline tests/access_test.go:354]: Maybe we could also refactor the creation of the user here. Just to have it in one place.

**Review by xpivarc — APPROVED** [review]

File: tests/access_test.go (hunk 1 above)
Comment (rmohr) [reply tests/access_test.go:354]: Tried a few things to find a useful pattern for a common function. I did not find a beneficial pattern. There is no common cleanup or creation flow with the other tests in vm_test.go which use this and the user is also created and deleted through the commandline. Want to keep it close to the tests for now.

Comment (xpivarc) [reply tests/access_test.go:354]: I was trying to point out that we have 
```
Context("should fail without admin rights for the project", func() {
			BeforeEach(func() {
				By("Ensuring the cluster has new test user")
				stdOut, stdErr, err := tests.RunCommandWithNS("", k8sClient, "create", "user", testUser)
				Expect(err).ToNot(HaveOccurred(), "ERR: %s", stdOut+stdErr)
				stdOut, stdErr, err = tests.RunCommandWithNS("", k8sClient, "project", util.NamespaceTestDefault)
				Expect(err).ToNot(HaveOccurred(), "ERR: %s", stdOut+stdErr)
			})
			AfterEach(func() {
				stdOut, stdErr, err := tests.RunCommandWithNS("", k8sClient, "delete", "user", testUser)
				Expect(err).ToNot(HaveOccurred(), "ERR: %s", stdOut+stdErr)
			})
```

and below it
 ```
Context("should succeed with admin rights for the project", func() {
			BeforeEach(func() {
				By("Ensuring the cluster has new test user")
				stdOut, stdErr, err := tests.RunCommandWithNS("", k8sClient, "create", "user", testUser)
				Expect(err).ToNot(HaveOccurred(), "ERR: %s", stdOut+stdErr)
				By("Ensuring user has the admin rights for the test namespace project")
				// This is ussually done in backgroung when creating new user with login and by creating new project by that user
				stdOut, stdErr, err = tests.RunCommandWithNS("", k8sClient, "adm", "policy", "add-role-to-user", "-n", util.NamespaceTestDefault, "admin", testUser)
				Expect(err).ToNot(HaveOccurred(), "ERR: %s", stdOut+stdErr)
				stdOut, stdErr, err = tests.RunCommandWithNS("", k8sClient, "project", util.NamespaceTestDefault)
				Expect(err).ToNot(HaveOccurred(), "ERR: %s", stdOut+stdErr)
			})
			AfterEach(func() {
				stdOut, stdErr, err := tests.RunCommandWithNS("", k8sClient, "delete", "user", testUser)
				Expect(err).ToNot(HaveOccurred(), "ERR: %s", stdOut+stdErr)
			})
```

Both contexts do create/delete the user eventually + something more. The user is more connected to the `Describe` imo and therefore I was suggesting to move closer to it. Just to clarify my comment.

Comment (rmohr) [reply tests/access_test.go:354]: oh, I misunderstood that. Will check it again.

/hold

**Review by dhiller — APPROVED** [review]

File: tests/access_test.go (hunk 1 above)
Comment (rmohr) [reply tests/access_test.go:354]: Done.

**Review by dhiller — APPROVED** [review]

**Pull request merged**